### PR TITLE
Update information about imagepruner object

### PR DESCRIPTION
**Pull request by dmage**
This is a cherry-pick of #23824 + updates for 4.5.

### CODE REVIEW

File: modules/pruning-images.adoc [hunk 1]
```diff
@@ -25,20 +20,19 @@ When the Pruning Custom Resource is deleted, the pruning `CronJob` and its relat
 [source,yaml]
 ----
 spec:
-  schedule: "0 0 * * *" <1>
+  schedule: 0 0 * * * <1>
   suspend: false <2>
   keepTagRevisions: 3 <3>
-  keepYoungerThan: 60m <4>
+  keepYoungerThanDuration: 60m <4>
```
Comment (dmage) [inline modules/pruning-images.adoc:26]: In 4.4 we have keepYoungerThan that is an integer in nanoseconds. It's deprecated in 4.5 and the new field `keepYoungerThanDuration` with normal values (strings with suffixes, like `1h`, `60m`, or `1h30m`) should be used instead.

Should docs mention the deprecated field?

File: modules/pruning-images.adoc [hunk 2]
```diff
@@ -55,29 +49,28 @@ status:
     reason: Succeeded
     message: "Most recent image pruning job succeeded."
 ----
-<1> `schedule`: `CronJob` formatted schedule, defaults to daily at midnight for new clusters. This is a required field.
-<2> `suspend`: If set to `true`, the `CronJob` running pruning is suspended. This is an optional field, and it defaults to `false`.
-<3> `keepTagRevisions`: The number of revisions per tag to keep. This is an optional field, and it defaults to `3` if not set.
-<4> `keepYoungerThan`: Retain images younger than this duration. This is an optional field, and it defaults `60m` if not set.
+<1> `schedule`: `CronJob` formatted schedule. This is an optional field, default is daily at midnight.
+<2> `suspend`: If set to `true`, the `CronJob` running pruning is suspended. This is an optional field, default is `false`. The initial value on new clusters is `false`.
```
Comment (dmage) [inline modules/pruning-images.adoc:53]: This differs from 4.4. On 4.4 the pruner config is bootstrapped with `suspend: true`.

File: modules/pruning-images.adoc (hunk 1 above)
Comment (bmcelvee) [reply modules/pruning-images.adoc:26]: That's a good question. I'm looking into it

**Comment by bmcelvee** [comment]
@wzheng1 if these changes look good to you, I will go ahead with the merge and cherry pick. Thanks!

**Comment by wzheng1** [comment]
Should we also explain:  "keepYoungerThan" will also be tolerated in 4.5, but if "keepYoungerThanDuration" is defined, "keepYoungerThan" will be overlapped. This information is useful for user who is using an upgraded cluster.

**Comment by bmcelvee** [comment]
Thanks, @wzheng1! I initially thought it was not necessary, but that is a good point. Since the bulk of these changes apply to 4.5+, I am leaning towards merging this one as is and cherry picking it to master and 4.6, then opening a follow-up PR to 4.5 specifically to add the `keepYoungerThan` explanation. 

@dmage does that make sense?

**Comment by vikram-redhat** [comment]
@dmage bump

**Comment by dmage** [comment]
@wzheng1 updated, please take a look

**Comment by simonkrenger** [comment]
@wzheng1 Requesting an update on this PR

**Comment by wzheng1** [comment]
Acked by QE!

**Comment by bmcelvee** [comment]
Thanks, all! Merging now.

**Comment by vikram-redhat** [comment]
@bmcelvee does this need going into 4.6 and 4.7 as well?

See this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1916082

**Comment by bmcelvee** [comment]
> @bmcelvee does this need going into 4.6 and 4.7 as well?
> 
> See this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1916082

That's a good question. @dmage do you think it would be appropriate to bring these changes forward? If so, I can make it happen.

**Comment by dmage** [comment]
@bmcelvee yes, please. I don't know why I made them only for 4.4 and 4.5.

**Pull request merged**